tools/update_requirements.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
```

Comment:
Does this script really update the freezed versions?
As far as I can tell, the container installs packages according to `requirements-freeze.txt`. Then runs a `pip freeze`, which should give the exact same requirements file. Then this script just takes that same file and overwrites the `tools/requirements-freeze.txt` with the exact same file. Or am I missing something?

Reply:
>Does this script really update the freezed versions?

The version would not change a lot of times, and that's the intention of the CR, to keep things freezed. But it could change. Following are the cases when this gets updated - 

* The first version of ```requirements-freeze``` has just the base packages (some 14) but after running  pip install commands run on container we get some 42 pip vended packages to lock. So we copy/overwrite them to requirements-freeze.txt. And this becomes our first git version if requirements-freeze.txt. 

* Let's say we added a new package dependency in requirements-freeze.txt, any new additions to package list should also get reflected in requirements-freeze.txt which is pulled from container.

* Same thing applies when we upgrade packages versions. 

Essentially it gives us more control to pick which versions we want to pick as part of new diff we see in the requirements-freeze.txt.

Reply:
I see. I was under the impression that this script should also perform an upgrade of the pip packages.
Isn't there something we could do so that the upgrade process would not be manual. With the current state of the PR, we would have to manually edit the requirements file for every single upgraded package.

Something like `./update_requirements.sh --upgrade `?

Reply:
As a side note, I would also rename the scripts to include `pip` in their name. `update_requirements` does not say much about what the requirements refer to. Maybe `update_pip_requirements`?

Reply:
> I see. I was under the impression that this script should also perform an upgrade of the pip packages.
> Isn't there something we could do so that the upgrade process would not be manual. With the current state of the PR, we would have to manually edit the requirements file for every single upgraded package.
> 
> Something like `./update_requirements.sh --upgrade `?

Right now upgrades are not supported, it would be manual, but I don't think it would be lot of pain. However we can make the support for upgrades as well later. 

One easy way to run upgrades would be to remove all the versions from requirements-freeze.txt and the new updated file will have upgraded versions.

Reply:
> As a side note, I would also rename the scripts to include `pip` in their name. `update_requirements` does not say much about what the requirements refer to. Maybe `update_pip_requirements`?

sounds good. done.